lib/modules/manager/docker-compose/schema.ts

```diff
@@ -14,13 +15,48 @@ const DockerComposeService = z.object({
 });
 
 const DockerComposeFileV1 = z.record(DockerComposeService);
-const DockerComposeFileModern = z.object({
-  // compose does not use this strictly, so we shouldn't be either
-  // https://docs.docker.com/compose/compose-file/04-version-and-name/#version-top-level-element
-  version: z.string().optional(),
-  services: z.record(DockerComposeService),
-});
+const DockerComposeFileModern = z
+  .object({
+    // compose does not use this strictly, so we shouldn't be either
+    // https://docs.docker.com/compose/compose-file/04-version-and-name/#version-top-level-element
+    version: z.string().optional(),
+    services: z.record(DockerComposeService),
+  })
+  .catchall(z.unknown()) // allow unknown fields
```

Comment:
i don't like that. why not simply add extensions to the root schema?

Reply:
`extensions` is a name I have given to collect all of them in one object
They can have any name as long as they `begin` with `x-` prefix hence we cannot use them directly as field.

Reply:
can we define the root schema like:
```ts
type DExtensionKey = `x-${string}`
const isValidDKey = (key: string): key is DExtensionKey => key.startsWith("x-")
const DCFile = z.record(z.string().refine(isValidDKey)).extend(
  z.object({
    version: z.string().optional(),
    services: z.record(DockerComposeService),
  })
);
```

Reply:
- https://github.com/colinhacks/zod/issues/2195

Reply:
☝️

Reply:
It asks to use `extend` with `record` which is not correct as per the Zod documentation. And, the suggested approach as per their AI is to use `catchall` as well

Reply:
maybe simply add a filter to only allow keys with `x-` 🤔 .

if not possible add a comment to the code that we need a newer (v4?) zod to make this better.

Reply:
Kinda doing something with the `catchall` in the current implemenation